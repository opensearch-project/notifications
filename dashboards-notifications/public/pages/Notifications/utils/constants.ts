/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];
