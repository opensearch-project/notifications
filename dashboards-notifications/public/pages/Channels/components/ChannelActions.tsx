/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiButton, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
import React, { useContext, useState } from 'react';
import { SERVER_DELAY } from '../../../../common';
import { ChannelItemType } from '../../../../models/interfaces';
import { CoreServicesContext } from '../../../components/coreServices';
import { ModalConsumer } from '../../../components/Modal';
import { ServicesContext } from '../../../services';
import { ROUTES } from '../../../utils/constants';
import { DeleteChannelModal } from './modals/DeleteChannelModal';
import { MuteChannelModal } from './modals/MuteChannelModal';

interface ChannelActionsParams {
  label: string;
  disabled: boolean;
  modal?: React.ReactNode;
  modalParams?: object;
  href?: string;
  action?: () => void;
}

interface ChannelActionsProps {
  selected: ChannelItemType[];
  setSelected: (items: ChannelItemType[]) => void;
  items: ChannelItemType[];
  setItems: (items: ChannelItemType[]) => void;
  refresh: () => void;
}

export function ChannelActions(props: ChannelActionsProps) {
  const coreContext = useContext(CoreServicesContext)!;
  const servicesContext = useContext(ServicesContext)!;
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);

  const actions: ChannelActionsParams[] = [
    {
      label: 'Edit',
      disabled: props.selected.length !== 1,
      href: `#${ROUTES.EDIT_CHANNEL}/${props.selected[0]?.config_id}`,
    },
    {
      label: 'Delete',
      disabled: props.selected.length === 0,
      modal: DeleteChannelModal,
      modalParams: { refresh: props.refresh },
    },
    {
      label: 'Mute',
      disabled: props.selected.length !== 1 || !props.selected[0].is_enabled,
      modal: MuteChannelModal,
      modalParams: { refresh: props.refresh, setSelected: props.setSelected },
    },
    {
      label: 'Unmute',
      disabled: props.selected.length !== 1 || props.selected[0].is_enabled,
      action: async () => {
        const channel = { ...props.selected[0], is_enabled: true };
        servicesContext.notificationService
          .updateConfig(channel.config_id, channel)
          .then((resp) => {
            coreContext.notifications.toasts.addSuccess(
              `Channel ${channel.name} successfully unmuted.`
            );
            props.setSelected([channel]);
            setTimeout(() => props.refresh(), SERVER_DELAY);
          })
          .catch((error) => {
            coreContext.notifications.toasts.addError(error, {
              title: 'Failed to unmute channel',
            });
          });
      },
    },
  ];

  return (
    <ModalConsumer>
      {({ onShow }) => (
        <EuiPopover
          panelPaddingSize="none"
          button={
            <EuiButton
              iconType="arrowDown"
              iconSide="right"
              disabled={props.selected.length === 0}
              onClick={() => setIsPopoverOpen(!isPopoverOpen)}
            >
              Actions
            </EuiButton>
          }
          isOpen={isPopoverOpen}
          closePopover={() => setIsPopoverOpen(false)}
        >
          {actions.map((params) => (
            <EuiContextMenuItem
              key={params.label}
              disabled={params.disabled}
              onClick={() => {
                setIsPopoverOpen(false);
                if (params.modal) {
                  onShow(params.modal, {
                    selected: props.selected,
                    ...(params.modalParams || {}),
                  });
                }
                if (params.href) location.assign(params.href);
                if (params.action) params.action();
              }}
            >
              {params.label}
            </EuiContextMenuItem>
          ))}
        </EuiPopover>
      )}
    </ModalConsumer>
  );
}
