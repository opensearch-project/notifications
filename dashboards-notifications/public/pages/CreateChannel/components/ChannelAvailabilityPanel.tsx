/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiCheckboxGroup,
  EuiCheckboxGroupOption,
  EuiFormRow,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import React from 'react';
import { ContentPanel } from '../../../components/ContentPanel';
import { NOTIFICATION_SOURCE } from '../../../utils/constants';

interface ChannelAvailabilityPanelProps {
  sourceCheckboxIdToSelectedMap: { [x: string]: boolean };
  setSourceCheckboxIdToSelectedMap: (map: { [x: string]: boolean }) => void;
}

export function ChannelAvailabilityPanel(props: ChannelAvailabilityPanelProps) {
  const sourceOptions: EuiCheckboxGroupOption[] = Object.entries(
    NOTIFICATION_SOURCE
  ).map(([key, value]) => ({
    id: key,
    label: value,
  }));

  return (
    <ContentPanel
      bodyStyles={{ padding: 'initial' }}
      title="Availability"
      titleSize="s"
    >
      <EuiFormRow label="Notification source">
        <>
          <EuiText size="xs" color="subdued">
            Select the features that can use this channel.
          </EuiText>
          <EuiSpacer size="s" />
          <EuiCheckboxGroup
            options={sourceOptions}
            idToSelectedMap={props.sourceCheckboxIdToSelectedMap}
            onChange={(optionId: string) => {
              props.setSourceCheckboxIdToSelectedMap({
                ...props.sourceCheckboxIdToSelectedMap,
                ...{
                  [optionId]: !props.sourceCheckboxIdToSelectedMap[optionId],
                },
              });
            }}
          />
        </>
      </EuiFormRow>
    </ContentPanel>
  );
}
