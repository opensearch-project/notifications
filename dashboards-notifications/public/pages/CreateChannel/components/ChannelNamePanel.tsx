/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiFieldText, EuiFormRow, EuiText, EuiTextArea } from '@elastic/eui';
import React, { useContext } from 'react';
import { ContentPanel } from '../../../components/ContentPanel';
import { CreateChannelContext } from '../CreateChannel';
import { validateChannelName } from '../utils/validationHelper';

interface ChannelNamePanelProps {
  name: string;
  setName: (name: string) => void;
  description: string;
  setDescription: (description: string) => void;
}

export function ChannelNamePanel(props: ChannelNamePanelProps) {
  const context = useContext(CreateChannelContext)!;
  return (
    <>
      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title="Name and description"
        titleSize="s"
      >
        <EuiFormRow
          label="Name"
          error="Name is required."
          isInvalid={context.inputErrors.name}
        >
          <EuiFieldText
            placeholder="Enter channel name"
            value={props.name}
            onChange={(e) => props.setName(e.target.value)}
            onBlur={() => {
              const error = validateChannelName(props.name);
              if (error !== context.inputErrors.name) {
                context.setInputErrors({
                  ...context.inputErrors,
                  name: error,
                });
              }
            }}
          />
        </EuiFormRow>
        <EuiFormRow
          label={
            <span>
              Description - <i style={{ fontWeight: 'normal' }}>optional</i>
            </span>
          }
        >
          <>
            <EuiText size="xs" color="subdued">
              Describe the purpose of the channel.
            </EuiText>
            <EuiTextArea
              placeholder="Describe the channel"
              style={{ height: '2.8rem' }}
              value={props.description}
              onChange={(e) => props.setDescription(e.target.value)}
            />
          </>
        </EuiFormRow>
      </ContentPanel>
    </>
  );
}
