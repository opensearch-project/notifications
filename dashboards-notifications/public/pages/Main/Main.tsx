/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiPage, EuiPageBody, EuiPageSideBar, EuiSideNav } from '@elastic/eui';
import React, { Component, createContext } from 'react';
import { Redirect, Route, RouteComponentProps, Switch } from 'react-router-dom';
import { CoreStart } from '../../../../../src/core/public';
import { CoreServicesConsumer } from '../../components/coreServices';
import { ModalProvider, ModalRoot } from '../../components/Modal';
import { BrowserServices } from '../../models/interfaces';
import { ServicesConsumer, ServicesContext } from '../../services/services';
import { CHANNEL_TYPE, ROUTES } from '../../utils/constants';
import { Channels } from '../Channels/Channels';
import { ChannelDetails } from '../Channels/components/details/ChannelDetails';
import { CreateChannel } from '../CreateChannel/CreateChannel';
import { CreateRecipientGroup } from '../Emails/CreateRecipientGroup';
import { CreateSender } from '../Emails/CreateSender';
import { EmailGroups } from '../Emails/EmailGroups';
import Notifications from '../Notifications';

enum Navigation {
  Notifications = 'Notifications',
  Dashboard = 'Dashboard',
  Channels = 'Channels',
  EmailGroups = 'Email groups',
}

enum Pathname {
  Notifications = '/notifications',
  Channels = '/channels',
}

interface MainProps extends RouteComponentProps {}

export interface MainState {
  availableFeatures: Partial<typeof CHANNEL_TYPE>;
  tooltipSupport: boolean; // if true, IAM role for SNS is optional and helper text should be available
}

export const MainContext = createContext<MainState | null>(null);

export default class Main extends Component<MainProps, MainState> {
  static contextType = ServicesContext;

  constructor(props: MainProps) {
    super(props);
    this.state = {
      availableFeatures: CHANNEL_TYPE,
      tooltipSupport: false,
    };
  }

  async componentDidMount() {
    const serverFeatures = await this.context.notificationService.getServerFeatures();
    if (serverFeatures != null)
      this.setState({
        availableFeatures: serverFeatures.availableFeatures,
        tooltipSupport: serverFeatures.tooltipSupport,
      });
  }

  render() {
    const {
      location: { pathname },
    } = this.props;

    const sideNav = [
      {
        name: Navigation.Notifications,
        id: 0,
        href: `#${Pathname.Notifications}`,
        items: [
          {
            name: Navigation.Dashboard,
            id: 1,
            href: `#${Pathname.Notifications}`,
            isSelected: pathname === Pathname.Notifications,
          },
          {
            name: Navigation.Channels,
            id: 2,
            href: `#${Pathname.Channels}`,
            isSelected: pathname === Pathname.Channels,
          },
          {
            name: Navigation.EmailGroups,
            id: 3,
            href: `#${ROUTES.EMAIL_GROUPS}`,
            isSelected: pathname === ROUTES.EMAIL_GROUPS,
          },
        ],
      },
    ];
    return (
      <CoreServicesConsumer>
        {(core: CoreStart | null) =>
          core && (
            <ServicesConsumer>
              {(services: BrowserServices | null) =>
                services && (
                  <MainContext.Provider value={this.state}>
                    <ModalProvider>
                      <ModalRoot services={services} />
                      <EuiPage>
                        {/*Hide side navigation bar when creating or editing rollup job*/}
                        {pathname !== ROUTES.CREATE_CHANNEL &&
                          !pathname.startsWith(ROUTES.EDIT_CHANNEL) &&
                          !pathname.startsWith(ROUTES.CHANNEL_DETAILS) &&
                          pathname !== ROUTES.CREATE_SENDER &&
                          !pathname.startsWith(ROUTES.EDIT_SENDER) &&
                          pathname !== ROUTES.CREATE_RECIPIENT_GROUP &&
                          !pathname.startsWith(ROUTES.EDIT_RECIPIENT_GROUP) && (
                            <EuiPageSideBar style={{ minWidth: 150 }}>
                              <EuiSideNav
                                style={{ width: 150 }}
                                items={sideNav}
                              />
                            </EuiPageSideBar>
                          )}
                        <EuiPageBody>
                          <Switch>
                            <Route
                              path={ROUTES.CREATE_CHANNEL}
                              render={(props: RouteComponentProps) => (
                                <CreateChannel {...props} />
                              )}
                            />
                            <Route
                              path={`${ROUTES.EDIT_CHANNEL}/:id`}
                              render={(
                                props: RouteComponentProps<{ id: string }>
                              ) => <CreateChannel {...props} edit={true} />}
                            />
                            <Route
                              path={`${ROUTES.CHANNEL_DETAILS}/:id`}
                              render={(
                                props: RouteComponentProps<{ id: string }>
                              ) => <ChannelDetails {...props} />}
                            />
                            <Route
                              path={ROUTES.CHANNELS}
                              render={(props: RouteComponentProps) => (
                                <Channels
                                  {...props}
                                  notificationService={
                                    services.notificationService
                                  }
                                />
                              )}
                            />
                            <Route
                              path={ROUTES.NOTIFICATIONS}
                              render={(props: RouteComponentProps) => (
                                <Notifications
                                  {...props}
                                  services={services}
                                  mainProps={this.state}
                                />
                              )}
                            />
                            <Route
                              path={ROUTES.EMAIL_GROUPS}
                              render={(props: RouteComponentProps) => (
                                <EmailGroups {...props} />
                              )}
                            />
                            <Route
                              path={ROUTES.CREATE_SENDER}
                              render={(props: RouteComponentProps) => (
                                <CreateSender {...props} />
                              )}
                            />
                            <Route
                              path={`${ROUTES.EDIT_SENDER}/:id`}
                              render={(props: RouteComponentProps) => (
                                <CreateSender {...props} edit={true} />
                              )}
                            />
                            <Route
                              path={ROUTES.CREATE_RECIPIENT_GROUP}
                              render={(props: RouteComponentProps) => (
                                <CreateRecipientGroup {...props} />
                              )}
                            />
                            <Route
                              path={`${ROUTES.EDIT_RECIPIENT_GROUP}/:id`}
                              render={(props: RouteComponentProps) => (
                                <CreateRecipientGroup {...props} edit={true} />
                              )}
                            />
                            <Redirect from="/" to={ROUTES.NOTIFICATIONS} />
                          </Switch>
                        </EuiPageBody>
                      </EuiPage>
                    </ModalProvider>
                  </MainContext.Provider>
                )
              }
            </ServicesConsumer>
          )
        }
      </CoreServicesConsumer>
    );
  }
}
