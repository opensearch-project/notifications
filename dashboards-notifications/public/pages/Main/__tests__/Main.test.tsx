/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {
  HashRouter as Router,
  Route,
  RouteComponentProps,
} from 'react-router-dom';
import Main from '..';
import {
  coreServicesMock,
  notificationServiceMock,
} from '../../../../test/mocks/serviceMock';
import { CoreServicesContext } from '../../../components/coreServices';
import { ServicesContext } from '../../../services';
import { ROUTES } from '../../../utils/constants';

describe('<Main /> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders the component', () => {
    const mockProps = {
      location: { search: '', pathname: ROUTES.NOTIFICATIONS },
      match: { params: { id: 'test' } },
    };
    const utils = render(
      <Router>
        <Route
          render={(props) => (
            <ServicesContext.Provider value={notificationServiceMock}>
              <CoreServicesContext.Provider value={coreServicesMock}>
                <Main {...(mockProps as RouteComponentProps<{ id: string }>)} />
              </CoreServicesContext.Provider>
            </ServicesContext.Provider>
          )}
        />
      </Router>
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
