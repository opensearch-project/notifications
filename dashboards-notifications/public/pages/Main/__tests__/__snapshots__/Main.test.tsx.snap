// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> spec renders the component 1`] = `
<div
  class="euiPage euiPage--paddingMedium euiPage--grow"
>
  <div
    class="euiPageSideBar"
    style="min-width: 155px;"
  >
    <nav
      class="euiSideNav"
      style="width: 155px;"
    >
      <div
        class="euiSideNav__content euiSideNav__contentMobile-xs euiSideNav__contentMobile-s"
        id="random_html_id"
      >
        <div
          class="euiSideNavItem euiSideNavItem--root euiSideNavItem--hasChildItems"
        >
          <a
            class="euiSideNavItemButton euiSideNavItemButton--isClickable"
            href="#/channels"
            rel="noreferrer"
          >
            <span
              class="euiSideNavItemButton__content"
            >
              <span
                class="euiSideNavItemButton__label euiSideNavItemButton__label--truncated"
                title="Notifications"
              >
                Notifications
              </span>
            </span>
          </a>
          <div
            class="euiSideNavItem__items"
          >
            <div
              class="euiSideNavItem euiSideNavItem--trunk"
            >
              <a
                class="euiSideNavItemButton euiSideNavItemButton--isClickable"
                href="#/channels"
                rel="noreferrer"
              >
                <span
                  class="euiSideNavItemButton__content"
                >
                  <span
                    class="euiSideNavItemButton__label euiSideNavItemButton__label--truncated"
                    title="Channels"
                  >
                    Channels
                  </span>
                </span>
              </a>
            </div>
            <div
              class="euiSideNavItem euiSideNavItem--trunk"
            >
              <a
                class="euiSideNavItemButton euiSideNavItemButton--isClickable"
                href="#/email-senders"
                rel="noreferrer"
              >
                <span
                  class="euiSideNavItemButton__content"
                >
                  <span
                    class="euiSideNavItemButton__label euiSideNavItemButton__label--truncated"
                    title="Email senders"
                  >
                    Email senders
                  </span>
                </span>
              </a>
            </div>
            <div
              class="euiSideNavItem euiSideNavItem--trunk"
            >
              <a
                class="euiSideNavItemButton euiSideNavItemButton--isClickable"
                href="#/email-recipient-groups"
                rel="noreferrer"
              >
                <span
                  class="euiSideNavItemButton__content"
                >
                  <span
                    class="euiSideNavItemButton__label euiSideNavItemButton__label--truncated"
                    title="Email recipient groups"
                  >
                    Email recipient groups
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="euiPageBody euiPageBody--borderRadiusNone"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        style="padding: 0px 0px;"
      >
        <div
          class="euiFlexItem"
        >
          <h3
            class="euiTitle euiTitle--medium"
          >
            Channels
            <span
              style="color: rgb(159, 159, 159); font-weight: 300;"
            >
               (0)
            </span>
          </h3>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem"
            >
              <div
                class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
              >
                <div
                  class="euiFlexItem euiFlexItem--flexGrowZero"
                >
                  <div
                    class="euiPopover euiPopover--anchorDownCenter"
                  >
                    <div
                      class="euiPopover__anchor"
                    >
                      <button
                        class="euiButton euiButton--primary euiButton-isDisabled"
                        disabled=""
                        type="button"
                      >
                        <span
                          class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                        >
                          <svg
                            aria-hidden="true"
                            class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                            focusable="false"
                            height="16"
                            role="img"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          />
                          <span
                            class="euiButton__text"
                          >
                            Actions
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="euiFlexItem euiFlexItem--flexGrowZero"
                >
                  <a
                    class="euiButton euiButton--primary euiButton--fill"
                    href="#/create-channel"
                    rel="noreferrer"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text"
                      >
                        Create channel
                      </span>
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
      />
      <div
        style="padding: initial;"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  class="euiFieldSearch euiFieldSearch--fullWidth"
                  placeholder="Search"
                  type="search"
                />
                <div
                  class="euiFormControlLayoutIcons"
                >
                  <span
                    class="euiFormControlLayoutCustomIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                      focusable="false"
                      height="16"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiFilterGroup"
            >
              <div
                class="euiPopover euiPopover--anchorDownCenter"
              >
                <div
                  class="euiPopover__anchor"
                >
                  <button
                    class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon euiFilterButton--noGrow"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      />
                      <span
                        class="euiButtonEmpty__text"
                      >
                        <span
                          class="euiFilterButton__textShift"
                          data-text="Status"
                          title="Status"
                        >
                          Status
                        </span>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="euiPopover euiPopover--anchorDownCenter"
              >
                <div
                  class="euiPopover__anchor"
                >
                  <button
                    class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon euiFilterButton--noGrow"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      />
                      <span
                        class="euiButtonEmpty__text"
                      >
                        <span
                          class="euiFilterButton__textShift"
                          data-text="Type"
                          title="Type"
                        >
                          Type
                        </span>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
        />
        <div
          class="euiBasicTable euiBasicTable-loading"
        >
          <div>
            <div
              class="euiTableHeaderMobile"
            >
              <div
                class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
              >
                <div
                  class="euiFlexItem euiFlexItem--flexGrowZero"
                >
                  <div
                    class="euiCheckbox"
                  >
                    <input
                      aria-label="Select all rows"
                      class="euiCheckbox__input"
                      disabled=""
                      id="_selection_column-checkbox_random_html_id"
                      type="checkbox"
                    />
                    <div
                      class="euiCheckbox__square"
                    />
                    <label
                      class="euiCheckbox__label"
                      for="_selection_column-checkbox_random_html_id"
                    >
                      Select all rows
                    </label>
                  </div>
                </div>
                <div
                  class="euiFlexItem euiFlexItem--flexGrowZero"
                >
                  <div
                    class="euiTableSortMobile"
                  >
                    <div
                      class="euiPopover euiPopover--anchorDownRight"
                    >
                      <div
                        class="euiPopover__anchor"
                      >
                        <button
                          class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                          type="button"
                        >
                          <span
                            class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                          >
                            <svg
                              aria-hidden="true"
                              class="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                              focusable="false"
                              height="16"
                              role="img"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            />
                            <span
                              class="euiButtonEmpty__text"
                            >
                              Sorting
                            </span>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <table
              class="euiTable euiTable--responsive euiTable--auto"
              id="random_html_id"
              tabindex="-1"
            >
              <caption
                class="euiScreenReaderOnly euiTableCaption"
              />
              <thead>
                <tr>
                  <th
                    class="euiTableHeaderCellCheckbox"
                    scope="col"
                  >
                    <div
                      class="euiTableCellContent"
                    >
                      <div
                        class="euiCheckbox euiCheckbox--inList euiCheckbox--noLabel"
                      >
                        <input
                          aria-label="Select all rows"
                          class="euiCheckbox__input"
                          data-test-subj="checkboxSelectAll"
                          disabled=""
                          id="_selection_column-checkbox_random_html_id"
                          type="checkbox"
                        />
                        <div
                          class="euiCheckbox__square"
                        />
                      </div>
                    </div>
                  </th>
                  <th
                    aria-live="polite"
                    aria-sort="ascending"
                    class="euiTableHeaderCell"
                    data-test-subj="tableHeaderCell_name_0"
                    role="columnheader"
                    scope="col"
                  >
                    <button
                      class="euiTableHeaderButton euiTableHeaderButton-isSorted"
                      data-test-subj="tableHeaderSortButton"
                      type="button"
                    >
                      <span
                        class="euiTableCellContent"
                      >
                        <span
                          class="euiTableCellContent__text"
                          title="Name"
                        >
                          Name
                        </span>
                        <svg
                          aria-hidden="true"
                          class="euiIcon euiIcon--medium euiIcon-isLoading euiTableSortIcon"
                          focusable="false"
                          height="16"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        />
                      </span>
                    </button>
                  </th>
                  <th
                    aria-live="polite"
                    aria-sort="none"
                    class="euiTableHeaderCell"
                    data-test-subj="tableHeaderCell_is_enabled_1"
                    role="columnheader"
                    scope="col"
                  >
                    <button
                      class="euiTableHeaderButton"
                      data-test-subj="tableHeaderSortButton"
                      type="button"
                    >
                      <span
                        class="euiTableCellContent"
                      >
                        <span
                          class="euiTableCellContent__text"
                          title="Notification status"
                        >
                          Notification status
                        </span>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-live="polite"
                    aria-sort="none"
                    class="euiTableHeaderCell"
                    data-test-subj="tableHeaderCell_config_type_2"
                    role="columnheader"
                    scope="col"
                  >
                    <button
                      class="euiTableHeaderButton"
                      data-test-subj="tableHeaderSortButton"
                      type="button"
                    >
                      <span
                        class="euiTableCellContent"
                      >
                        <span
                          class="euiTableCellContent__text"
                          title="Type"
                        >
                          Type
                        </span>
                      </span>
                    </button>
                  </th>
                  <th
                    aria-live="polite"
                    aria-sort="none"
                    class="euiTableHeaderCell"
                    data-test-subj="tableHeaderCell_description_3"
                    role="columnheader"
                    scope="col"
                  >
                    <button
                      class="euiTableHeaderButton"
                      data-test-subj="tableHeaderSortButton"
                      type="button"
                    >
                      <span
                        class="euiTableCellContent"
                      >
                        <span
                          class="euiTableCellContent__text"
                          title="Description"
                        >
                          Description
                        </span>
                      </span>
                    </button>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="euiTableRow"
                >
                  <td
                    class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                    colspan="5"
                  >
                    <div
                      class="euiTableCellContent euiTableCellContent--alignCenter"
                    >
                      <span
                        class="euiTableCellContent__text"
                      >
                        <div
                          class="euiEmptyPrompt"
                        >
                          <h2
                            class="euiTitle euiTitle--medium"
                          >
                            No channels to display
                          </h2>
                          <span
                            class="euiTextColor euiTextColor--subdued"
                          >
                            <div
                              class="euiSpacer euiSpacer--m"
                            />
                            <div
                              class="euiText euiText--medium"
                            >
                              To send or receive notifications, you will need to create a notification channel.
                            </div>
                          </span>
                          <div
                            class="euiSpacer euiSpacer--l"
                          />
                          <a
                            class="euiButton euiButton--primary"
                            href="#/create-channel"
                            rel="noreferrer"
                          >
                            <span
                              class="euiButtonContent euiButton__content"
                            >
                              <span
                                class="euiButton__text"
                              >
                                Create channel
                              </span>
                            </span>
                          </a>
                        </div>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
